File: assignment-2/q4/remfile.py
fname = input().strip() #Name of text file to be changed
foutn = input().strip() #Name of output text file
n = int(input()) #Number to be removed from the file

#Your code here
#Opening the input file in read mode
inputFile = open(fname, "r")

#Reading the file and making a list for it
inputRead = inputFile.read().split()

#Checking if the input file is empty
if not inputRead:
    print('The input file is empty')
    exit()

#if the file is not empty, do this
else:
    #Opening the Output file in write mode
    outputFile = open(foutn, 'w')

    #Looping over the input text and removing the words with the input Number
    #by adding it to the output file
    for i in inputRead:
        stringN = str(n)
        #check if the current index of the list is not equal to index value with the number
        #if it's not there, add the word to the output file
        if i.find(stringN) == -1:
            outputFile.write(i + " ")
        else:
            continue
    #Close the files
    inputFile.close()
    outputFile.close()

File: assignment-2/q4/remfile.c
#include <stdio.h>
#include <stdlib.h> // For exit()
#include <string.h>
int main(){
	char fname[100];//Name of text file to be changed
	char foutn[100];//Name of the output text file
	int n;//Number to be removed from the file
	fgets(fname, 100, stdin);
	fgets(foutn, 100, stdin);
	scanf("%d", &n);
	printf("%d\n", n);

	//Your code here

	//Declaring the a pointer of file type to both input and output files
  FILE *inputFile;
	FILE *outputFile;
  char c;

	//fgets retain the newline character, so we add one for input filename
	fname [ strcspn(fname, "\r\n") ] = 0;

 	//Opening the input file in read mode
	inputFile = fopen(fname, "r");

	//fgets retain the newline character, so we add one for output filename
	foutn [ strcspn(foutn, "\r\n") ] = 0;

	//Opening the output file in write mode
	outputFile = fopen(foutn, "w");

	if (inputFile == NULL) {
		printf("%s\n", "Cannot open file");
		exit(0);
	}

	c = fgetc(inputFile);
	while (c != EOF)
	{
			printf ("%c", c);
			c = fgetc(inputFile);
	}

	fclose(inputFile);
	fclose(outputFile);


	return 0;
}
